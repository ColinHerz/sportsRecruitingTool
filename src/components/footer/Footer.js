import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import React from "react";

import "./footer.scss";

const Footer = props => {
	const year = (new Date()).getFullYear();

	return (
		<footer>
			<div id="logo">
				<h1>Sporta</h1>
				<p>&copy; Sporta {year}</p>
			</div>

			{
				props.loggedIn ?
					<div id="links">
						<h2>Menu</h2>

						<ul>
							<li><Link to="/myevents/garbage/">Events</Link></li>
							<li><Link to="/scores/">Top Scores</Link></li>
							<li><Link to="/profile/garbage/">My Profile</Link></li>
							<li><button onClick={props.logOut}>Log Out</button></li>
						</ul>
					</div>
					:<div id="links">
						<h2>Get Started</h2>

						<ul>
							<li><button onClick={props.loggingIn}>Log In</button></li>
							<li><button onClick={props.registering}>Sign Up</button></li>
						</ul>
					</div>
			}
		</footer>
	);
};

Footer.propTypes = {
	loggingIn: PropTypes.func.isRequired,
	logOut: PropTypes.func.isRequired,
	registering: PropTypes.func.isRequired,
	loggedIn: PropTypes.bool.isRequired
};

export default Footer;
