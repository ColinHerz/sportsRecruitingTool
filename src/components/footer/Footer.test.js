/* eslint-env jest */

// MemoryRouter needed because header contains NavLinks, which need to
// be rendered within a Router
import { MemoryRouter } from "react-router-dom";
import { mount } from "../../../enzyme.js";
import React from "react";
import renderer from "react-test-renderer";

import Footer from "./Footer.js";

const genProps = loggedIn => {
	return {
		loggingIn: jest.fn(),
		registering: jest.fn(),
		loggedIn: loggedIn,
		logOut: jest.fn()
	};
};

beforeEach(() => {
	jest.resetModules();
});

describe(`Rendering`, () => {
	it(`renders correctly when not logged in`, () => {
		const props = genProps(false);

		const component = renderer.create(
			<MemoryRouter>
				<Footer {... props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it(`renders correctly when logged in`, () => {
		const props = genProps(true);

		const component = renderer.create(
			<MemoryRouter>
				<Footer {... props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe(`Functionality`, () => {
	it(`calls log in when log in button clicked`, () => {
		const props = genProps(false);

		const component = mount(
			<MemoryRouter>
				<Footer {... props} />
			</MemoryRouter>
		);

		component.find(`#log-in`).simulate(`click`);

		expect(props.loggingIn).toHaveBeenCalled();

		component.unmount();
	});

	it(`calls registering when register button clicked`, () => {
		const props = genProps(false);

		const component = mount(
			<MemoryRouter>
				<Footer {... props} />
			</MemoryRouter>
		);

		component.find(`#sign-up`).simulate(`click`);

		expect(props.registering).toHaveBeenCalled();

		component.unmount();
	});

	it(`calls log out when log out button clicked`, () => {
		const props = genProps(true);

		const component = mount(
			<MemoryRouter>
				<Footer {... props} />
			</MemoryRouter>
		);

		component.find(`button`).simulate(`click`);

		expect(props.logOut).toHaveBeenCalled();

		component.unmount();
	});
});
