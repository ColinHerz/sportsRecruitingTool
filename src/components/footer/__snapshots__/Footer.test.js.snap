// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly when logged in 1`] = `
<footer>
  <div
    id="logo"
  >
    <h1>
      Sporta
    </h1>
    <p>
      © Sporta 
      2020
    </p>
  </div>
  <div
    id="links"
  >
    <h2>
      Menu
    </h2>
    <ul>
      <li>
        <a
          href="/myevents/garbage/"
          onClick={[Function]}
        >
          Events
        </a>
      </li>
      <li>
        <a
          href="/scores/"
          onClick={[Function]}
        >
          Top Scores
        </a>
      </li>
      <li>
        <a
          href="/profile/garbage/"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
      <li>
        <button
          onClick={[MockFunction]}
        >
          Log Out
        </button>
      </li>
    </ul>
  </div>
</footer>
`;

exports[`Footer component renders correctly when not logged in 1`] = `
<footer>
  <div
    id="logo"
  >
    <h1>
      Sporta
    </h1>
    <p>
      © Sporta 
      2020
    </p>
  </div>
  <div
    id="links"
  >
    <h2>
      Get Started
    </h2>
    <ul>
      <li>
        <button
          onClick={[MockFunction]}
        >
          Log In
        </button>
      </li>
      <li>
        <button
          onClick={[MockFunction]}
        >
          Sign Up
        </button>
      </li>
    </ul>
  </div>
</footer>
`;
